src/CharacterDetails/CharacterDetails.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { getCharacterById } from '../services/fetch-utils';
+
+export default function CharacterDetails() {
+  const [character, setCharacter] = useState({});
+  const { id } = useParams();
+
+  useEffect(() => {
+    async function load(){
+      const character = await getCharacterById(id);
+      setCharacter(character);
+    }
+    load();
+  }, [id]);
+  return (
+    <div className='character-detail'>
+      <h2>{character.name} level {character.level}</h2>
+      <p>Strength: {character.str}</p>
+      <p>Dexterity: {character.dex}</p>
+      <p>Constitution: {character.con}</p>
+      <p>Intelligence: {character.int}</p>
+      <p>Wisdom: {character.wis}</p>
+      <p>Charisma: {character.cha}</p>
+    </div>
+  );
+}
```

Comment:
This page properly fetches based on `id` but we also want to be able to update the character with a form preloaded with the characters values.